engine/flink/avro-util/src/test/scala/pl/touk/nussknacker/engine/avro/schema/StringForcingDatumReaderSpec.scala

```diff
@@ -0,0 +1,63 @@
+package pl.touk.nussknacker.engine.avro.schema
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
+import org.apache.avro.generic.{GenericDatumWriter, GenericRecord}
+import org.apache.avro.io.{DecoderFactory, EncoderFactory}
+import org.apache.avro.Schema
+import org.scalatest.{EitherValues, FunSpec, Matchers}
+import pl.touk.nussknacker.engine.avro.{AvroStringSettingsInTests, AvroUtils}
+import pl.touk.nussknacker.engine.avro.encode.{BestEffortAvroEncoder, ValidationMode}
+
+
+class StringForcingDatumReaderSpec extends FunSpec with Matchers with EitherValues {
```

Comment:
It would be nice to see here not only encoding records via `BestEffortAvroEncoder` but also creating it from scratch using `new LogicalTypesGenericRecordBuilder(schema).set("foo", new Utf8(...)) ... / "")` with corner cases like: schema with default string field but not filled during building records. Will it work correctly? Or `new GenericRecord(otherRecord, deepCopy = true)` <- AFAIK it won't work and will be nice to show it in tests. Also would be nice to see what is inside record after encoding - even in round trips, I mean: `encodedRecord.get("foo") should be instanceOf[String]`

Reply:
`BestEffortAvroEncoder` was here only to create GenericRecord, actually it is not imporant how we create it. What's important here is reading it by DatumReader after serialization, and that is what I test.
I agree on the other hand, that it is worth writing test for provding default value from schema by LogicalTypesGenericRecordBuilder - it acctualy does use DatumReader (and in consequent StringForcingDatumReader)

`GenericData.Record::new(deepCopy=true)` also does not use DatumReader underneath, so I believe this change doesnt affect this operation